pennylane_forest/qpu.py

```diff
@@ -126,7 +120,7 @@ def expval(self, observable, wires, par):
                              ExperimentSetting(TensorProductState(), float(np.sqrt(1/2)) * sZ(qubit))]
             }
             # expectation values for single-qubit observables
-            if observable in ["PauliX", "PauliY", "PauliZ", "Identity", "Hadamard"]:
+            if observable.name in ["PauliX", "PauliY", "PauliZ", "Identity", "Hadamard"]:
```

Comment:
I realize that it is not part of this PR, but I find the code block below (in `for instr in self.program.instructions:`) to be... perplexing. It seems to do nothing but split out the instruction string, and then construct a copy of the original instruction from this. I think the whole thing could be replaced with 
```
prep_prog = Program([instr for instr in self.program.instructions if isinstance(instr, Gate)])
```
If one wants a copy of these instructions (e.g. for later mutation), then
```
prep_prog = Program([instr.out() for instr in self.program.instructions if isinstance(instr, Gate)])
```

Reply:
I think I missed this in the previous PR, but the main point here was to use `str_instr += f' {self.wiring[int(w)]}'`, though you could probably still use list comprehension (I just thought the logic would be less clear if I did). @josh146 I'm assuming, since `self.wiring[int(w)]` is replaced with just `int(w)` here, that PennyLane can now handle instructions on e.g. qubit 16 on say a 4q-lattice (and that I can specifically target the hypothetical qubit 16 in my instructions too)?

Reply:
Unfortunately, PennyLane doesn't yet support passing the specific qubit numbers (https://github.com/XanaduAI/pennylane/issues/423) but we would like to prioritize it after the upcoming release.

The reason the mapping has been removed from here is that the mapping is now taken into account in the inherited `QVMDevice` instead. This is because we noticed that the QVM now enforces using the actual qubit numbers, similarly to the QPU. (Previously, the QVM would accept `{0, 1, ..., N-1}`, but it's nice that it now aligns with how the QPU behaves 🙂.)